aten/src/ATen/BatchedFallback.cpp

```diff
@@ -318,7 +345,12 @@ void batchedTensorForLoopFallback(const c10::OperatorHandle& op, torch::jit::Sta
   auto output_shards_chunks = MatrixRef<Tensor>(output_shards, num_batches);
   for (int64_t return_idx = 0; return_idx < num_returns; ++return_idx) {
     auto shards = output_shards_chunks[return_idx];
-    auto flat_output = at::stack(shards);
+    auto flat_output = safeStack(shards);
+    // See NOTE [vmap through backward and undefined grad]
+    if (!flat_output.defined()) {
```

Comment:
What's the deal with undefined Tensors and batch dimensions? Should they have some?

Reply:
BatchedTensor cannot hold an undefined tensor.

Reply:
Sounds good!